perun/collect/complexity/symbols.py

```diff
@@ -174,11 +177,10 @@ def _get_symbols(executable_path, columns):
     # Convert the columns list to string parameter
     columns_str = ','.join(str(column) for column in columns)
 
-    # # readelf -sW exec | awk '$4 == "FUNC" && $7 != "UND"' | awk '{$2=$2};1' | cut -d' ' -f columns
-    cmd = "readelf -sW " + executable_path + \
-          " | awk '$4 == \"FUNC\" && $7 != \"UND\"' | awk '{$2=$2};1' | cut -d' ' -f " + columns_str
-    ps = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
-    symbols = ps.communicate()[0]
+    cmd = ("readelf -sW " + executable_path + " | awk '$4 == \"FUNC\" && $7 != \"UND\"' | "
```

Comment:
I am wondering, how many times is this subprocess called? Just once?

Because, when I was playing with memory-collector I ran into huge performance degradation because of subsequent calls to addr2line and had to cache the results to obtain reasonable times.